include Ivy Local into resolving compiler interface and compiler bridge

Currently it requires `publishM2` instead of `publishLocal`.

A variation of this issue is that currently it requires `publishM2` of util because util-interface is needed for compile-interface.

## steps

1. `publishLocal` util with SHA1 version.
2. change incrementalcompiler's build to rely on the SHA1 version of util.
3. run test on incrementalcompiler after `publishM2`ing the incrementalcompiler JARs

## problem

One problem is that when the compiler fails to find the interface, it seems to keep going, and then it fails to compile the bridge:

```scala
/var/folders/7g/l96y5q310h90xsz109qqjpmc0000gn/T/sbt_10b7c08a/xsbt/CompilerInterface.scala:6: error: object Logger is not a member of package xsbti
import xsbti.{ AnalysisCallback, Logger, Problem, Reporter, Severity }
       ^
```

## expectation

I can `publishLocal` both util and incrementalcompiler.

--- a/build.sbt
+++ b/build.sbt
@@ -241,8 +241,8 @@ lazy val publishBridgesAndTest = Command.args("publishBridgesAndTest", "<version
   val version = args mkString ""
   val compilerInterfaceID = compilerInterface.id
   val compilerBridgeID = compilerBridge.id
-  val test = s"$compilerInterfaceID/publishM2" :: s"plz $version incrementalcompilerRoot/test" :: s"plz $version incrementalcompilerRoot/scripted" :: state
-  (scalaVersions map (v => s"plz $v $compilerBridgeID/publishM2") foldRight test) { _ :: _ }
+  val test = s"$compilerInterfaceID/publishLocal" :: s"plz $version incrementalcompilerRoot/test" :: s"plz $version incrementalcompilerRoot/scripted" :: state
+  (scalaVersions map (v => s"plz $v $compilerBridgeID/publishLocal") foldRight test) { _ :: _ }
 }
 
 lazy val otherRootSettings = Seq(

--- a/internal/incrementalcompiler-ivy-integration/src/test/scala/sbt/internal/inc/BridgeProviderSpecification.scala
+++ b/internal/incrementalcompiler-ivy-integration/src/test/scala/sbt/internal/inc/BridgeProviderSpecification.scala
@@ -9,7 +9,7 @@ import sbt.internal.inc.classpath.ClasspathUtilities
 import sbt.internal.librarymanagement.{ JsonUtil, ComponentManager, BaseIvySpecification }
 import sbt.io.IO
 import sbt.io.Path._
-import sbt.librarymanagement.{ ModuleID, UpdateOptions, Resolver }
+import sbt.librarymanagement.{ ModuleID, UpdateOptions, Resolver, Patterns, FileRepository, DefaultMavenRepository }
 import sbt.util.{ Logger, Level }
 import xsbti.{ ComponentProvider, GlobalLock }
 
@@ -19,7 +19,12 @@ import xsbti.{ ComponentProvider, GlobalLock }
 abstract class BridgeProviderSpecification extends BaseIvySpecification {
   log.setLevel(Level.Warn)
 
-  override def resolvers: Seq[Resolver] = super.resolvers ++ Seq(Resolver.mavenLocal, Resolver.jcenterRepo)
+  def realLocal: Resolver =
+    {
+      val pList = s"$${user.home}/.ivy2/local/${Resolver.localBasePattern}" :: Nil
+      FileRepository("local", Resolver.defaultFileConfiguration, Patterns(pList, pList, false))
+    }
+  override def resolvers: Seq[Resolver] = Seq(realLocal, DefaultMavenRepository)
   private val ivyConfiguration = mkIvyConfiguration(UpdateOptions())
 
   def getCompilerBridge(targetDir: File, log: Logger, scalaVersion: String): File = {

